File: Domain/BeSpoked.BLL/SalesTeam/Entities/SalesPerson.cs
using BeSpoked.Common.EntityService.Entities;
using BeSpoked.Sales.Entities;

namespace BeSpoked.SalesTeam.Entities;

public record SalesPerson : Entity
{
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    //public Address
    public required string Phone { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime? TerminationDate { get; set; }
    public required string Manager { get; set; }
    
    public virtual ICollection<Sale> Sales { get; set; } = new HashSet<Sale>();
};

File: WebApp/ClientApp/src/Entities/SalesPerson.ts
export interface SalesPerson {
    id: string;
    firstName: string;
    lastName: string;
    phone: string;
    startDate: Date;
    terminationDate: Date | null
    manager: string;
}

export type ServerSalesPerson = Omit<SalesPerson, "startDate" | "terminationDate"> & { startDate: string, terminationDate: string | null}

export function deserializeSalesPerson(serverResponse:ServerSalesPerson):SalesPerson{
    return {
        ...serverResponse,
        startDate: new Date(serverResponse.startDate),
        terminationDate: serverResponse.terminationDate ? new Date(serverResponse.startDate) : null
    }
}